src/natpmp.cpp

```diff
@@ -278,7 +277,8 @@ void natpmp::try_next_mapping(int const i)
 
 	auto const m = std::find_if(
 		m_mappings.begin(), m_mappings.end()
-		, [] (mapping_t const& ma) { return ma.act != mapping_t::action::none; });
+		, [] (mapping_t const& ma) { return ma.act != mapping_t::action::none &&
```

Comment:
since we clear the action everywhere, presumably we don't need to check the protocol here, right?